api/services/authentication/implementations/otp.js

```diff
@@ -0,0 +1,27 @@
+const { Op } = require('sequelize');
+const DB = require('models');
+const { v4 } = require('uuid');
+const AuthStrategyInterface = require('../interface');
+
+class OtpAuthStrategy extends AuthStrategyInterface {
+  constructor(app) {
+    super(app);
+    this._redis = app.getService('redis').client;
+  }
+
+  async authenticate({ phone, otp }) {
+    const correctOtp = await this._redis.get(phone);
+    if (correctOtp == otp) {
```

Comment:
`==` is it safe to use here? can convert both to string, trim, lowercase and then compare?

Reply:
On a second thought, even converting to string, trimming and lowercasing would fail for the case of NULL. i.e. redis returns 'null'. That will be changed to string "null" automatically during comparison. If otp is also null, passed by user intentionally, the check would pass. 

Here's a silly solution
const correctOtp = await this._redis.get(phone) || uuid4();